src/github/copilotRemoteAgent/chatSessionContentBuilder.ts

```diff
@@ -0,0 +1,407 @@
+		} else {
+			sessionPrompt = await this.getFollowUpSessionPrompt(sessionIndex, timelineEvents, sessionPrompt);
+		}
+
+		// TODO: @rebornix, remove @copilot prefix from session prompt for now
+		sessionPrompt = sessionPrompt.replace(/@copilot\s*/gi, '').trim();
+		return sessionPrompt;
+	}
+
+	private async getFollowUpSessionPrompt(
+		sessionIndex: number,
+		timelineEvents: readonly TimelineEvent[],
+		defaultPrompt: string
+	): Promise<string> {
+		const copilotStartedEvents = timelineEvents
+			.filter((event): event is CopilotStartedEvent => event.event === EventType.CopilotStarted)
+			.sort((a, b) => new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime());
+
+		const copilotFinishedEvents = timelineEvents
+			.filter((event): event is CopilotFinishedEvent => event.event === EventType.CopilotFinished)
+			.sort((a, b) => new Date(a.createdAt).getTime() - new Date(b.createdAt).getTime());
+
+		Logger.appendLine(`Session ${sessionIndex}: Found ${copilotStartedEvents.length} CopilotStarted events and ${copilotFinishedEvents.length} CopilotFinished events`, this.loggerId);
+
+		const copilotStartedEvent = copilotStartedEvents[sessionIndex];
+		if (!copilotStartedEvent) {
+			Logger.appendLine(`Session ${sessionIndex}: No CopilotStarted event found at index ${sessionIndex}`, this.loggerId);
+			return defaultPrompt;
+		}
+
+		const currentSessionStartTime = new Date(copilotStartedEvent.createdAt).getTime();
+		const previousSessionEndTime = this.getPreviousSessionEndTime(sessionIndex, copilotFinishedEvents);
+
+		const relevantEvents = this.findRelevantTimelineEvents(timelineEvents, previousSessionEndTime, currentSessionStartTime);
+
+		const matchingEvent = relevantEvents[0];
+		if (matchingEvent) {
+			const prompt = this.extractPromptFromEvent(matchingEvent);
+			Logger.appendLine(`Session ${sessionIndex}: Found matching event - ${matchingEvent.event}`, this.loggerId);
+			return prompt;
+		} else {
+			Logger.appendLine(`Session ${sessionIndex}: No matching event found between times ${previousSessionEndTime} and ${currentSessionStartTime}`, this.loggerId);
+			Logger.appendLine(`Session ${sessionIndex}: Relevant events found: ${relevantEvents.length}`, this.loggerId);
+			return defaultPrompt;
+		}
+	}
+
+	private getPreviousSessionEndTime(sessionIndex: number, copilotFinishedEvents: CopilotFinishedEvent[]): number {
+		if (sessionIndex > 0 && copilotFinishedEvents[sessionIndex - 1]) {
+			return new Date(copilotFinishedEvents[sessionIndex - 1].createdAt).getTime();
+		}
+		return 0;
+	}
+
+	private findRelevantTimelineEvents(
+		timelineEvents: readonly TimelineEvent[],
+		previousSessionEndTime: number,
+		currentSessionStartTime: number
+	): TimelineEvent[] {
+		return timelineEvents
+			.filter(event => {
+				if (event.event !== EventType.Commented && event.event !== EventType.Reviewed) {
+					return false;
+				}
+
+				const eventTime = new Date(
```

Comment:
There's a utility for getting the event time!

https://github.com/microsoft/vscode-pull-request-github/blob/73e670b2a6996d1f5e957b354f0bc2af2ad375d9/src/github/utils.ts#L1060-L1061